src/twisted/web/test/test_wsgi.py

```diff
@@ -1941,6 +1943,68 @@ def cbRendered(ignored):
         return d
 
 
+    @inlineCallbacks
+    def test_pendingRequestOperationsRespectConnectionLoss(self):
+        """
+        If there are pending transport writes after
+        C{Request.notifyFinish} is called, the transport should
+        not be written to.
+        """
+
+        self.enableThreads()
+        trackedRequest = [None]
+
+        def applicationFactory():
+            def application(environ, startResponse):
+                startResponse('200 OK', [])
+                self.reactor.callFromThread(
+                    trackedRequest[0].connectionLost,
+                    Failure(ConnectionLost("No more connection")))
+                yield b'write'
+
+            return application
+
+        def requestFactory(*args, **kwargs):
+            # We need to manually close the request instantiated
+            # by self.lowLevelRender().
+            self.assertIsNone(trackedRequest[0])
+            trackedRequest[0] = Request(*args, **kwargs)
+            return trackedRequest[0]
+
+        # Keep a copy of _WSGIResponse.write before we patch it.
+        _write = _WSGIResponse.write
+
+        # Signal for the delayed write allowing it to proceed.
+        connectionLost = threading.Event()
+
+        # Result of the (delayed) write.
+        writeResult = Deferred()
+
+        def _WSGIResponse_write(self, data):
+            # Block write until transport is closed.
+            connectionLost.wait()
+
+            try:
+                r = _write(self, data)
+                self.reactor.callFromThread(writeResult.callback, None)
+                return r
+            except:
+                self.reactor.callFromThread(writeResult.errback, Failure())
```

Comment:
I feel that this code will be not executed in the tests and it has a very generic exception handling.

I think that it needs a comment and at least `except Exception` to not catch Ctrl+C or other higher level exceptions

Without a comment, someone later might see that it has no coverage and might decide to delete it.

I saw that without this, the test will block.

Reply:
Agreed. Added a note about the reasoning behind this.